department-of-veterans-affairs/abd-vro
# EP Merge App: paginate get merge jobs endpoint

## What was the problem?

Current `/merge` endpoint returns all merge jobs. It needs to be updated to return paginated responses with optional page offset, page size, and filter by job state.

Associated tickets or Slack threads:
- #2412 
- Stacked On:
  - https://github.com/department-of-veterans-affairs/abd-vro/pull/2375

## How does this fix it?[^1]

Makes the necessary updates to the endpoint. See the AC for user store #2412 . 

## How to test this PR
- pull code
- Unit Test:
  - from /domain-ee/ee-ep-merge-app run pytest
- Integration Tests:
  - run [base platform](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Docker-Compose#platform-base) 
  - from /domain-ee/ee-ep-merge-app run pytest ./integration
- End to End tests: 
  - run this [CI](https://github.com/department-of-veterans-affairs/abd-vro/actions/workflows/ee-ep-merge-end-to-end.yml) action to verify successful integration test


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
